client/client.go

```diff
@@ -563,24 +567,55 @@ func (c *client) GetURLs() []string {
 	return c.urls
 }
 
+// TsoReqAlloc defines how client create and reuse tsoRequest.
+type TsoReqAlloc interface {
```

Comment:
Too many interfaces 😢 AFAICS, all you want is to use different pool instances for allocating the object. How about export `NewPool` instead, and pass an extra `pool` argument?